src/python/pants/backend/kotlin/compile/kotlinc.py

```diff
@@ -83,6 +89,16 @@ async def compile_kotlin_source(
         ),
     )
 
+    plugins_ = await MultiGet(
+        Get(
+            KotlincPluginTargetsForTarget,
+            KotlincPluginsForTargetRequest(target, request.resolve.name),
+        )
+        for target in request.component.members
+    )
```

Comment:
Since a plugin can never actually be used outside the context of a CT (i.e.: if there is a cycle, you will always compile the entire cycle), `KotlincPluginsForTargetRequest` could probably take a `CT`.

Reply:
I took this code verbatim from the scalac plug-in code. I assume a similar comment applies there as well and should be updated?

Reply:
Yes. It's a nit though probably... shouldn't make an appreciable difference in runtime unless you have a lot of cycles.

Reply:
I'm inclined to land as-is and refactor both scala and kotlin in a follow-up.